### PR TITLE
Replace inappropriate terms

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed?

- Replace Mesos-related `slave` to `agent` to align with its [official documentation](http://mesos.apache.org/documentation/latest/architecture/).
- Replace MySQL-related `master-slave` to `source-replica`. ref: [MySQL Terminology Updates](https://mysqlhighavailability.com/mysql-terminology-updates/)
- Refine an item in FAQ

### CODE REVIEW

**Comment by ran-huang** [comment]
@dcalvin @zhangjinpeng1987 @WT-Liu PTAL, thanks!

File: content/docs/3.0/reference/faq.md [hunk 1]
```diff
@@ -144,9 +144,9 @@ This is because the address in the startup parameter has been registered in the
 
 To solve this problem, use the [`store delete`](https://github.com/pingcap/pd/tree/55db505e8f35e8ab4e00efd202beb27a8ecc40fb/tools/pd-ctl#store-delete--label--weight-store_id----jqquery-string) function to delete the previous store and then restart TiKV.
 
-## TiKV master and slave use the same compression algorithm, why the results are different?
+## TiKV leader replicas and follower replicas use the same compression algorithm, why the results are different?
```
Comment (zhangjinpeng1987) [inline content/docs/3.0/reference/faq.md:147]: s/results/the amount of disk space occupied

**Comment by WT-Liu** [comment]
LGTM

File: content/docs/4.0/reference/faq.md [hunk 2]
```diff
@@ -144,9 +144,9 @@ This is because the address in the startup parameter has been registered in the
 
 To solve this problem, use the [`store delete`](https://github.com/pingcap/pd/tree/55db505e8f35e8ab4e00efd202beb27a8ecc40fb/tools/pd-ctl#store-delete--label--weight-store_id----jqquery-string) function to delete the previous store and then restart TiKV.
 
-## TiKV master and slave use the same compression algorithm, why the results are different?
+## TiKV leader replicas and follower replicas use the same compression algorithm, why the amount of disk space occupied is different?
```
Comment (dcalvin) [inline content/docs/4.0/reference/faq.md:147]:
```suggestion
## TiKV leader replicas and follower replicas use the same compression algorithm. Why the amount of disk space occupied is different?
```

File: content/docs/dev/reference/faq.md [hunk 3]
```diff
@@ -144,9 +144,9 @@ This is because the address in the startup parameter has been registered in the
 
 To solve this problem, use the [`store delete`](https://github.com/pingcap/pd/tree/55db505e8f35e8ab4e00efd202beb27a8ecc40fb/tools/pd-ctl#store-delete--label--weight-store_id----jqquery-string) function to delete the previous store and then restart TiKV.
 
-## TiKV master and slave use the same compression algorithm, why the results are different?
+## TiKV leader replicas and follower replicas use the same compression algorithm, why the amount of disk space occupied is different?
```
Comment (dcalvin) [inline content/docs/dev/reference/faq.md:147]:
```suggestion
## TiKV leader replicas and follower replicas use the same compression algorithm. Why the amount of disk space occupied is different?
```

**Comment by ran-huang** [comment]
@dcalvin comments addressed. PTAL again~

**Pull request merged**